src/android/CodePushReportingManager.java

```diff
@@ -13,28 +12,7 @@
 
     private Activity cordovaActivity;
     private CodePushPreferences codePushPreferences;
-
-    /**
-     * Defines application statuses we use in reporting events from the native to the JS layer.
-     */
-    public static enum Status {
-        STORE_VERSION(0),
-        UPDATE_CONFIRMED(1),
-        UPDATE_ROLLED_BACK(2);
-
-        private int value;
-
-        Status(int i) {
-            this.value = i;
-        }
-
-        /**
-         * Returns the value associated with this enum.
-         */
-        public int getValue() {
-            return this.value;
-        }
-    }
+    private Boolean hasFailedReport = null;
```

Comment:
it seems like the only use of this is to avoid checking the preferences...but given that you don't report updates very often and this variable could get out of sync fairly easily i think it'd be better to remove this and just use codePushPreferences